tests/S3/Parser/ValidateResponseChecksumResultMutatorTest.php

```diff
@@ -25,25 +25,40 @@ class ValidateResponseChecksumResultMutatorTest extends TestCase
      * @dataProvider checksumCasesDataProvider
      * @param array $responseAlgorithms
      * @param array $checksumHeadersReturned
-     * @param string $expectedChecksum
-     *
+     * @param string|null $expectedChecksumAlgorithm
+     * @param array $config
+     * @param string $operation
+     * @param string|null $checksumMode
      * @return void
      */
-    public function testValidatesChoosesRightChecksum(
+    public function testChecksumValidation(
         array $responseAlgorithms,
         array $checksumHeadersReturned,
-        ?string $expectedChecksumAlgorithm
+        ?string $expectedChecksumAlgorithm,
+        array $config,
+        string $operation,
+        ?string $checksumMode
     ) {
+        if (!empty($responseAlgorithms)
+            && $responseAlgorithms[0] === 'CRC32C'
+            && !extension_loaded('awscrt')
+        ) {
+            $this->markTestSkipped();
+        }
+
         $s3Client = $this->getTestS3ClientWithResponseAlgorithms(
             'GetObject',
```

Comment:
I think you also need to pass the operation here. Instead of `GetObject` it should be `$operation`.

Reply:
I think I intended to remove the `$operation` param at some point because we were only using `GetObject`, but I'll leave that there in case we want to test others in the future.